deployment/environment.go

```diff
@@ -76,6 +76,7 @@ type Environment struct {
 	Chains            map[uint64]Chain
 	NodeIDs           []string
 	Offchain          OffchainClient
+	Context           context.Context
```

Comment:
> Do not store Contexts inside a struct type; instead, pass a Context explicitly to each function that needs it. The Context should be the first parameter, typically named ctx

https://pkg.go.dev/context

Reply:
We are deviating from the norm here, because this Environment struct is effectively a context-object (in the broad sense), and is in the first position in all changesets, and is the way we plumb through all of the infrastructure, among which we are adding context.Context.

While this may be the norm for call stacks, it isn't part of the signature, and Environment was intended to be an extensible bit of plumbing, to avoid having to change a lot of signatures.

Reply:
Per a LONG thread, I've adjusted this to store a `func() context.Context` instead as the means to plumb it through.